Harvey-OS/harvey
# Reintroduce libventi

commit 275f0b42411259f8078477cf24f2b338d47c6ab2
Author: Dan Cross <cross@gajendra.net>
Date:   Thu Apr 20 16:19:04 2017 +0000

    Really expose warnings

    But also make it so that we can continue building.
    By exposing them, we see them so we can fix them.

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 331ffebb5394eb0768cea498e43a2d2c98829e9d
Author: Sevki Hasirci <s@sevki.org>
Date:   Mon Feb 22 00:55:22 2016 +0100

    just the files

commit 400364d361586e548bd7c08d8cdcb64a4bc69e90
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 13:53:16 2016 -0700

    fixfmt: with this last round of changes I can boot.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit 48090c023db7a9d0f4f36f3b270dc30080c24ce5
Author: ron minnich <rminnich@gmail.com>
Date:   Sun Feb 14 05:49:03 2016 +0100

    Revert "build: added new build files"
    Does not work at all. Can't build a working system with it yet.

    Let's take this slower, ok? Not break it all next time.

    This reverts commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12.

    Change-Id: I91fce92249dd9d7bf3ca731a124c61a499382dd2

commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12
Author: Sevki <sevki@spotify.com>
Date:   Wed Jan 27 12:46:31 2016 +0100

    build: added new build files

    kernel compiles and crashes

    closes #37

    to test use

            go get -u sevki.org/build/cmd/build
            build //sys/src/9/amd64:harvey
    or

            build -v //sys/src/9/amd64:harvey

    inside a editor like emacs or acme.

    If you are building on OS X you should also have something like .build
    file at the root of the project and add something like

            CC: gcc
            TOOLPREFIX: x86_64-elf-

    Change-Id: Ib6f6156eee1936ceb5f8b0bfa64ed45589195c31
    Signed-off-by: Sevki <sevki@spotify.com>

commit 5d943862678b8b33c0514a99862998c182b3aeb5
Author: Álvaro Jurado <elbingmiss@gmail.com>
Date:   Fri Aug 26 15:18:42 2016 +0200

    Another round of %u modifier

    This fixes tar, fossil and probably some other.
    All source reviewed. I hope this will be last fix about this.

    Signed-off-by: Álvaro Jurado <elbingmiss@gmail.com>

commit 6533db6114e97d5687bdb6a0f94136c98d009408
Author: Dan Cross <cross@gajendra.net>
Date:   Mon May 1 00:00:16 2017 +0000

    Remove trailing whitespace from source files.

    This is a trivial cleanup (done with a script, Ron!).

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 789474da8c5122316f7ee093a02e84a4aed24ad2
Author: Dan Cross <cross@gajendra.net>
Date:   Thu Apr 20 17:59:21 2017 +0000

    Kill a couple of thousand warnings.

    These are the easy ones. I've got a few hundred left.

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 7955c22c6ac664b6482dcfc8f8aeca45dbb8a49c
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Sun Jul 31 21:15:21 2016 -0700

    fmt: another round of format fixes for %#ul*x to remove the u

    Remove the u.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit 9fba51d999489cf1bf2adf3a35c2260caebf0c32
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 13:09:10 2016 -0700

    fixfmt: fix all usages of %u so that u is a type, not a modifier

    The rest of the world uses %u to mean unsigned decimal. Plan 9
    uses it as a modifier, which is arguably better, but what can we do?
    You don't always get to pick the standard.

    This change modifies libc/fmt so that %u is unsigned decimal, and
    modifies most uses of it so the kernel almost boots. Almost. Obviously
    we didn't get them all yet.

    It hangs in ipconfig.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit a9d13806e2923195accce1ddbd0caa6e18ce386b
Author: fuchicar <rafita.fernandez@gmail.com>
Date:   Fri Mar 10 12:33:27 2017 +0100

    Removed ifndef _MPINT from libsec.h

    This is not Harvey style. Removing it implies add #include <mp.h> before each #include <libsec.h>

    Signed-off-by: fuchicar <rafita.fernandez@gmail.com>

commit b5a9f6c2a79429df122dc96df2fb6095ba9c8b49
Author: Hank Donnay <hdonnay@gmail.com>
Date:   Sun Aug 2 12:59:20 2015 -0500

    Fix compiler warnings in libventi

    cvt.h: cast 48- and 64-bit integers into uint64_t instead of uint32_t

    file.c: 'oe' seems to have been a pointer to the original VtEntry, but
    was only assigned to and never looked at again.

    Change-Id: I175d48d5f396028579af006ab4a7201c97743c17
    Signed-off-by: Hank Donnay <hdonnay@gmail.com>

commit ba14fa27776ef2371a42607ec04bf50287fbd0be
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Fri Nov 13 08:20:41 2015 -0800

    New json format per Giacomo's ideas.

    Instead of this:
    {
            "Name": "a",
            ...
    }

    Do this:
    {
            "Name": {
                    ...
            }
    }

    This is way better. Now the name is visible at the top level of any json viewer, we can
    have multiple stanzas per file, the name is way more easy to find, ... the advantages are legion.

    This includes a change to build.go but not to preen.

    This now works fine, BUILD all produces a working kernel and userland. But someone else needs to verify it.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

    Change-Id: I9b1cf2597e582895b8f3ab127bc9c1d77de96cbd

commit c58c6c510d5875253d66b4d26d9ae9025f9191f3
Author: Dan Cross <cross@gajendra.net>
Date:   Mon May 1 00:39:33 2017 +0000

    Remove blank lines at end of file.

    Remove a bunch of blank lines at the end of files.

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit cac72d4555626b4e1693a68113baf37f3e30c790
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 14:15:44 2016 -0700

    fixfmt: The final final set of changes.

    I missed %lud and %llud somehow. Still boots and
    output from du -s makes more sense.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit e087b93e8188160b096e82e5c9bf27657118847f
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 13:48:15 2016 -0700

    fixfmt: more fixes. Also, always include ratrace in ramfs

    It makes debugging when booting from rc far, far easier.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>
